compiler/noirc_frontend/src/elaborator/types.rs

```diff
@@ -28,72 +32,85 @@ use crate::{
         UnaryOp, UnresolvedType, UnresolvedTypeData,
     },
     node_interner::{DefinitionKind, ExprId, GlobalId, TraitId, TraitImplKind, TraitMethodId},
-    Generics, Type, TypeBinding, TypeVariable, TypeVariableKind,
+    Generics, ResolvedGeneric, Type, TypeBinding, TypeKind, TypeVariable, TypeVariableKind,
 };
 
 use super::{lints, Elaborator};
 
 impl<'context> Elaborator<'context> {
-    /// Translates an UnresolvedType to a Type
+    /// Translates an UnresolvedType to a Type with a `TypeKind::Normal`
     pub(super) fn resolve_type(&mut self, typ: UnresolvedType) -> Type {
-        let span = typ.span;
-        let resolved_type = self.resolve_type_inner(typ);
-        if resolved_type.is_nested_slice() {
```

Comment:
a bit surprised to see the `is_nested_slice` check moved to `resolve_type_inner`: were there cases where the slice was only nested at the end of it? also, could this simply be moved to after the call to `resolve_type_inner`? currently, it appears to be re-run recursively

Reply:
I just moved it as to not repeat the check whenever we make a call to `resolve_type_inner` as we now call `resolve_type_inner` twice. 

> currently, it appears to be re-run recursively

This is a good point. Also, I guess `resolve_numeric_type` should only really be called when we want to mkae sure we have a numeric named generic, so we might not even need to repeat the check. I'll switch it back to be after the `resolve_type_inner` call.

Reply:
moved the check back to `resolve_type`